src/Core/SortCursor.h

```diff
@@ -164,17 +180,36 @@ struct SortCursor : SortCursorHelper<SortCursor>
     /// The specified row of this cursor is greater than the specified row of another cursor.
     bool ALWAYS_INLINE greaterAt(const SortCursor & rhs, size_t lhs_pos, size_t rhs_pos) const
     {
+#if USE_EMBEDDED_COMPILER
+        if (impl->desc.compiled_sort_description && rhs.impl->desc.compiled_sort_description)
+        {
+            assert(impl->raw_sort_columns_data.size() == rhs.impl->raw_sort_columns_data.size());
+
+            auto sort_description_func_typed = reinterpret_cast<JITSortDescriptionFunc>(impl->desc.compiled_sort_description);
+            int res = sort_description_func_typed(lhs_pos, rhs_pos, impl->raw_sort_columns_data.data(), rhs.impl->raw_sort_columns_data.data()); /// NOLINT
+
+            if (res > 0)
+                return true;
+            if (res < 0)
+                return false;
+
+            return impl->order > rhs.impl->order;
```

Comment:
just a matter of taste
```suggestion
            if (res != 0)
                return res;
            return impl->order > rhs.impl->order;
```

Reply:
It is not the same. res can be 1, or -1, if we convert them to bool they both will be false. We can write res > 0, but current implementation is readable enought.